mobile/lib/shared/providers/app_state.provider.dart

```diff
@@ -8,6 +23,70 @@ enum AppStateEnum {
   detached,
 }
 
-final appStateProvider = StateProvider<AppStateEnum>((ref) {
-  return AppStateEnum.active;
+class AppStateProvider extends StateNotifier<AppStateEnum> {
+  final Ref ref;
+
+  AppStateProvider(this.ref) : super(AppStateEnum.active);
+
+  void updateAppState(AppStateEnum appState) {
+    state = appState;
+  }
+
+  AppStateEnum getAppState() {
+    return state;
+  }
+
+  void handleAppResume() {
+    state = AppStateEnum.resumed;
+
+    var isAuthenticated = ref.watch(authenticationProvider).isAuthenticated;
+    final permission = ref.watch(galleryPermissionNotifier);
+
+    // Needs to be logged in and have gallery permissions
+    if (isAuthenticated && (permission.isGranted || permission.isLimited)) {
+      ref.read(backupProvider.notifier).resumeBackup();
+      ref.read(backgroundServiceProvider).resumeServiceIfEnabled();
+      ref.watch(assetProvider.notifier).getAllAsset();
```

Comment:
Why do we have the mix between read and watch here?

Reply:
Those were refactored to this file from `main.dart`. I changed nothing in terms of reading or watching a provider. The only change I made is inside the `handleAppInactivity` method

Reply:
Thank you, there was actually a place where I should be watching for changes in the Notifier inside `handleAppInactivity` which I was reading instead. Changed it to `watch` now :)